### ISSUE TITLE
SDL 1.2.15 and OS X issues

### ISSUE THREAD

**Issue by clintbellanger** [clintbellanger/flare-engine#336]
It looks like the OSX 10.7+ fullscreen issue was patched in SDL 1.2.15. But compiling with 1.2.15 on OS X reveals various issues.
### Very bad performance on blits.

HD resolution framerates drop from easy 30 to about 15 on my dev machine.

Performance tools show the SDL function SDL_RemoveTimer taking up a lot of time, but it might be a red herring -- it's called from SDL_LowerBlit (in turn called from SDL_UpperBlit and mainly from the Iso render functions of Flare).

The symptoms remind me of blitting between surfaces with different pixel formats, but I don't know if that's what is happening here.
### Fullscreen fixed on later versions of OS X but causes a crash on 10.6

We have to verify that SDL 1.2.15 fullscreen is supposed to work on OSX 10.6 and previous versions. Or do we have to create one OSX build for 10.6 and before, and a separate one for 10.7+? Hopefully not.

There may be something dumb we're doing. E.g. we may have to create a new SDL_Surface *screen after going fullscreen if we're not already. 
### SetGamma causing black (0,0,0) to change to a weird yellow/green color

This happens as soon as we open the Config menu. There may be something incorrect we're doing with the gamma functions, e.g. assuming certain values/calculations instead of using the built-in gamma utility functions to calculate those.

Commenting out all the SDL_SetGamma calls in the engine "fixes" the issue.
### Pink-transparency surfaces with the new SDL_image

This may be loosely related. We tried updating to the new SDL_image. Disabling high-quality textures and blitting them shows errors. The pink (255, 0, 255) isn't transparent.

I think I was able to make it transparent by adding an alpha channel to one of the pink images, even if I didn't actually change any of the pixels from their completely opaque look. I think one of the cleanup functions we use  when loading images is supposed to create an alpha channel if one is needed.

Maybe we're using those functions wrong. Maybe we need to create something like a loadImage() wrapper that does all those optimizations the same way every time -- once we determine the actual correct way to do that.

**Comment by igorko** [comment]
Did anyone try on SDL2. Are these issues present there?

**Comment by ludoza** [comment]
- flare-engine commit 21bf1c8b05af53b82ae41e24dfc0a433e847b08e
- flare-game commit 8ff2888394144fab22dfead6a4aaef9c1e2c757d
- sdl2: stable 2.0.3 (bottled), HEAD
- OS X 10.10.2 (14C1514)
- [MacBook Air (13-inch, Mid 2012)](https://en.wikipedia.org/wiki/MacBook_Air#Specifications)

I'm experiencing mouse issues, without hardware_cursor=1 the mouse will have a strange offset, and jump if you click.

With hardware_cursor=1 the mouse positioning is fixed but there is still issues with the mouse just "clicking".

After setting no_mouse=1 and getting into a game(with some struggle due to above mouse issues) I could walk around a bit the game crashed after a couple of minutes.

**tl;dr;** So for me it is not playable due to mouse issues(really makes the game unplayable) and the crash.

I will poke around a bit, and if I must test anything let me know.

**Comment by ludoza** [comment]
Played the Empyrean Campaign up until the grave yard with spawning zombies.

- Got 58 fps the whole time(is this limited somewhere?)
- Fullscreen is not true fullscreen. It just maximizes the window and still show the system menu and window headings

But other then the fixed mouse issue, I find the game playable on OS X.

**Comment by dorkster** [comment]
@ludoza 

- The game is locked at 60 fps. The problem only seems to be with the displayed fps value (rounding issue maybe?)
- Just to clarify, are you toggling fullscreen through Flare's configuration menu? Or are you pressing the green button in the window heading? As I understand it, the latter is supposed to fullscreen applications in recent OS X versions, but it's likely that SDL might not do that on its own.

**Comment by dorkster** [comment]
Can anyone (maybe @clintbellanger, since you opened this issue) give an update on the state of running Flare on OSX? I think SDL 2.0.4 should handle fullscreen correctly now, but I'd like verification.

**Comment by dorkster** [comment]
Going to link this thread here: https://opengameart.org/forumtopic/mac-version-not-working

Seems SDL 2 was the right way to go.
